flink-ml-lib/src/test/java/org/apache/flink/ml/feature/stringindexer/StringIndexerTest.java

```diff
@@ -0,0 +1,319 @@
+        try {
+            output = stringIndexer.fit(trainTable).transform(predictTable)[0];
+            IteratorUtils.toList(tEnv.toDataStream(output).executeAndCollect());
+            fail();
+        } catch (Exception e) {
+            assertEquals(
+                    "The input contains unseen string: e. "
+                            + "To handle unseen strings, set Param handleInvalid to keep.",
+                    e.getCause().getCause().getCause().getCause().getCause().getMessage());
+        }
+    }
+
+    @Test
+    public void testFitAndPredict() throws Exception {
+        StringIndexer stringIndexer =
+                new StringIndexer()
+                        .setInputCols("inputCol1", "inputCol2")
+                        .setOutputCols("outputCol1", "outputCol2")
+                        .setStringOrderType(StringIndexerParams.ALPHABET_ASC_ORDER)
+                        .setHandleInvalid(StringIndexerParams.KEEP_INVALID);
+        Table output = stringIndexer.fit(trainTable).transform(predictTable)[0];
+        List<Row> predictedResult =
+                IteratorUtils.toList(tEnv.toDataStream(output).executeAndCollect());
+        verifyPredictionResult(expectedAlphabeticAscPredictData, predictedResult);
+    }
+
+    @Test
+    public void testSaveLoadAndPredict() throws Exception {
+        StringIndexer stringIndexer =
+                new StringIndexer()
+                        .setInputCols("inputCol1", "inputCol2")
+                        .setOutputCols("outputCol1", "outputCol2")
+                        .setStringOrderType(StringIndexerParams.ALPHABET_ASC_ORDER)
+                        .setHandleInvalid(StringIndexerParams.KEEP_INVALID);
+        stringIndexer =
+                StageTestUtils.saveAndReload(
+                        env, stringIndexer, tempFolder.newFolder().getAbsolutePath());
+        StringIndexerModel model = stringIndexer.fit(trainTable);
+        model = StageTestUtils.saveAndReload(env, model, tempFolder.newFolder().getAbsolutePath());
+        assertEquals(
+                Collections.singletonList("stringsArray"),
+                model.getModelData()[0].getResolvedSchema().getColumnNames());
+        Table output = model.transform(predictTable)[0];
+        List<Row> predictedResult =
+                IteratorUtils.toList(tEnv.toDataStream(output).executeAndCollect());
+        verifyPredictionResult(expectedAlphabeticAscPredictData, predictedResult);
+    }
+
+    @Test
+    public void testGetModelData() throws Exception {
+        StringIndexerModel model =
+                new StringIndexer()
+                        .setInputCols("inputCol1", "inputCol2")
+                        .setOutputCols("outputCol1", "outputCol2")
+                        .setStringOrderType(StringIndexerParams.ALPHABET_ASC_ORDER)
+                        .fit(trainTable);
+        StringIndexerModelData modelData =
+                StringIndexerModelData.getModelDataStream(model.getModelData()[0])
```

Comment:
Thanks for the comment. The check for schema is added in StringIndexerTest#testGetModelData.

It is not added in `IndexToStringModelTest#testGetModelData()` because the model data is generated by `StringIndexer`. So I believe that the check for schema is already covered by StringIndexerTest#testGetModelData.

What do you think?

Reply:
Sounds good. Thanks.